contracts/modules/Wallet/VestingEscrowWallet.sol

```diff
@@ -0,0 +1,525 @@
+
+    struct Template {
+        uint256 numberOfTokens;
+        uint256 duration;
+        uint256 frequency;
+    }
+
+    enum State {CREATED, STARTED, COMPLETED}
+
+    ERC20 public token;
+    address public treasury;
+    uint256 public unassignedTokens;
+
+    mapping (address => Data) public dataMap;
+    address[] public beneficiaries;
+
+    Template[] public templates;
+
+    event AddSchedule(
+        address _beneficiary,
+        uint256 _numberOfTokens,
+        uint256 _duration,
+        uint256 _frequency,
+        uint256 _startTime,
+        uint256 _timestamp
+    );
+    event EditSchedule(
+        address _beneficiary,
+        uint256 _index,
+        uint256 _numberOfTokens,
+        uint256 _duration,
+        uint256 _frequency,
+        uint256 _startTime,
+        uint256 _timestamp
+    );
+    event RevokeSchedules(address _beneficiary, uint256 _timestamp);
+    event RevokeSchedule(address _beneficiary, uint256 _index, uint256 _timestamp);
+    event DepositTokens(uint256 _numberOfTokens, uint256 _timestamp);
+    event SendToTreasury(uint256 _numberOfTokens, uint256 _timestamp);
+    event SendTokens(address _beneficiary, uint256 _numberOfTokens, uint256 _timestamp);
+    event AddTemplate(uint256 _numberOfTokens, uint256 _duration, uint256 _frequency, uint256 _timestamp);
+    event RemoveTemplate(uint256 _index, uint256 _timestamp);
+
+    /**
+     * @notice Constructor
+     * @param _securityToken Address of the security token
+     * @param _polyAddress Address of the polytoken
+     */
+    constructor (address _securityToken, address _polyAddress)
+    public
+    Module(_securityToken, _polyAddress)
+    {
+    }
+
+    /**
+     * @notice Function used to initialize the different variables
+     * @param _treasury Address of the treasury
+     */
+    function configure(address _treasury, address _token) public onlyFactory {
+        require(_treasury != address(0), "Invalid address");
+        treasury = _treasury;
+        token = ERC20(_token);
+    }
+
+    /**
+     * @notice This function returns the signature of the configure function
+     */
+    function getInitFunction() public pure returns (bytes4) {
+        return bytes4(keccak256("configure(address,address)"));
+    }
+
+    /**
+     * @notice Used to deposit tokens from treasury
+     */
+    function depositTokens(uint256 _numberOfTokens) external withPerm(ADMIN) {
+        require(_numberOfTokens > 0, "Should be greater than zero");
+        token.safeTransferFrom(treasury, this, _numberOfTokens);
+        unassignedTokens = unassignedTokens.add(_numberOfTokens);
+        /*solium-disable-next-line security/no-block-members*/
+        emit DepositTokens(_numberOfTokens, now);
+    }
+
+    /**
+     * @notice Sends unassigned tokens to treasury
+     */
+    function sendToTreasury() external withPerm(ADMIN) {
+        uint256 amount = unassignedTokens;
+        unassignedTokens = 0;
+        token.safeTransfer(treasury, amount);
```

Comment:
Same here

Reply:
Shouldn't we prevent everyone to sending tokens to a treasury?
If someone will constantly invoke this method it will be hard to create a schedule.

Reply:
I am pointing here related to `safeTransfer`(use ST interface not ERC20). not related to permissions, permissions are good